File: intranet2copia/intranet2copia/src/main/java/intra/intranet2copia/Entidades/Usuario.java
package intra.intranet2copia.Entidades;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import java.util.List;

@Entity
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nombre;
    private String email;
    private String contrasena;

    @OneToMany(mappedBy = "usuario", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<SistemaArduino> sistemasArduino;

    // Getters y setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getContrasena() {
        return contrasena;
    }

    public void setContrasena(String contrasena) {
        this.contrasena = contrasena;
    }

    public List<SistemaArduino> getSistemasArduino() {
        return sistemasArduino;
    }

    public void setSistemasArduino(List<SistemaArduino> sistemasArduino) {
        this.sistemasArduino = sistemasArduino;
    }
}

File: apiv1admin - copia/Usuarios/Usuario.kt
package com.example.apiv1admin.Usuarios

import com.google.gson.annotations.SerializedName
import intra.intranet2copia.Entidades.SistemaArduino

data class Usuario(
    @SerializedName("id") val id: Long,
    @SerializedName("nombre") val nombre: String,
    @SerializedName("email") val email: String,
    @SerializedName("contrasena") val contrasena: String,
    @SerializedName("sistemasArduino") val sistemasArduino: List<SistemaArduino>?
)
